unlock-app/src/components/creator/CreatorLock.js

```diff
@@ -3,48 +3,76 @@ import UnlockPropTypes from '../../propTypes'
 import LockIconBar from './lock/LockIconBar'
 import CreatorLockConfirming from './lock/CreatorLockConfirming'
 import Icon from '../lock/Icon'
+import EmbedCodeSnippet from './lock/EmbedCodeSnippet'
 import Duration from '../helpers/Duration'
 import Balance from '../helpers/Balance'
+import Buttons from '../interface/buttons/lock'
 import styled from 'styled-components'
 
-export function CreatorLock({ lock, status = 'deployed' }) {
-  // Some sanitization of strings to display
-  let name = lock.name || 'New Lock'
-  let outstandingKeys = lock.maxNumberOfKeys - lock.outstandingKeys || 0
-  let lockComponentStatusBlock
-
-  if (status === 'deployed') { // the transaction was mined and confirmed at least 12 times
-    lockComponentStatusBlock = (<LockIconBarContainer>
-      <LockIconBar lock={lock} className={'lock-icons'} />
-    </LockIconBarContainer>)
-  } else {
-    lockComponentStatusBlock = <CreatorLockConfirming lock={lock} status={status} />
+export class CreatorLock extends React.Component {
+  constructor (props, context) {
+    super(props)
+    this.state = {
+      showEmbedCode: false,
+    }
+    this.toggleEmbedCode = this.toggleEmbedCode.bind(this)
+  }
+
+  toggleEmbedCode() {
+    this.setState({
+      showEmbedCode: !this.state.showEmbedCode,
+    })
   }
 
-  // TODO add USD values to lock
-  // TODO add all-time balance to lock
-  return (
-    <LockRow>
-      <Icon lock={lock} address={lock.address} />
-      <LockName>
-        {name}
-        <LockAddress>{lock.address}</LockAddress>
-      </LockName>
-      <LockDuration>
-        <Duration seconds={lock.expirationDuration} />
-      </LockDuration>
-      <LockKeys>{outstandingKeys}/{lock.maxNumberOfKeys}</LockKeys>
-      <Balance amount={lock.keyPrice} />
-      <Balance amount={lock.balance} />
-      {lockComponentStatusBlock}
-    </LockRow>
-  )
+  render() {
+    // TODO add USD values to lock
+    // TODO add all-time balance to lock
+
+    // Some sanitization of strings to display
+    let name = this.props.lock.name || 'New Lock'
+    let outstandingKeys = this.props.lock.maxNumberOfKeys - this.props.lock.outstandingKeys || 0
+    let lockComponentStatusBlock
+
+    if (this.props.status === 'deployed') { // the transaction was mined and confirmed at least 12 times
+      // The className below is required to hide / reveal the icon bar
```

Comment:
hum, I really do not think this is a great idea. we can do `hover` inside the styled component right?